src/slopes.jl

```diff
@@ -0,0 +1,198 @@
+# Reference : Dietmar Ratz - An Optimized Interval Slope Arithmetic and its Application
+using IntervalArithmetic, ForwardDiff
+import Base: +, -, *, /, ^, sqrt, exp, log, sin, cos, tan, asin, acos, atan
+import IntervalArithmetic: mid, interval
+
+function slope(f::Function, x::Interval, c::Real)
+    try
+        f(slope_var(x, c)).s
+    catch y
+        if isa(y, MethodError)
+            ForwardDiff.derivative(f, x)
+        end
+    end
+end
+
+struct Slope{T}
+    x::Interval{T}
+    c::Interval{T}
+    s::Interval{T}
+    Slope{T}(a, b, c) where T = new(a, b, c)
+    Slope{T}(c) where T = Slope{T}(c, c, 0)
```

Comment:
This can be an outer constructor.

Inner constructors are necessary only if you need to impose something particular when an object is created (eg that something cannot be negative).